File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {

        var dict = new Dictionary<int,ListNode>();
        var counter = 1;
        var current = head;

        do {
            dict.Add(counter, current);
            current = current.next;
            counter++;
        }
        while(current is not null);

        if(!dict.ContainsKey(counter -1 - n)){
            return head.next;
        }
        var previousToTarget = dict[counter -1 - n];

        if(!dict.ContainsKey(counter +1 - n)){
            previousToTarget.next = null;
            return head;
        }
        var nextToTarget = dict[counter +1 - n];

        previousToTarget.next = nextToTarget;


        return head;
        
    }
}

File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {ListNode}
     */
    reverseList(head) {

        let prev = null
        let curr = head
        
        if(!head) return head
        if(!head.next) return head

        let next = head.next

        while (curr) {
            next = curr.next;
            curr.next = prev;

            prev = curr;
            curr = next;
        }

        return prev


    }
}
